### ISSUE TITLE
tpot.regressor_config_dict affect training speed

### ISSUE THREAD

**Issue by huangqingyi-code** [EpistasisLab/tpot#1183]
Hi tpot team,
    Thank you for this great library. I have a question as follows.
     use tpot's regressor_config_dict,training speed is fast。When I reset the regressor_config the training speed is very slow.
code_fast:
from tpot.config import regressor_config_dict
regressor_config_dict['xgboost.XGBRegressor'] = {
        'n_estimators': [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000],
        'max_depth': range(3, 100),
        'learning_rate': [0.01, 0.03, 0.05, 0.07, 0.1, 0.3, 0.5],
        'subsample': np.arange(0, 1, 0.05),
        'gamma': np.arange(0, 10, 0.5),
        'min_child_weight': range(1, 10),
    }
tpot = TPOTRegressor(generations=20, population_size=50, verbosity=2, random_state=42,
                         n_jobs=-1, max_time_mins=5, config_dict=regressor_config_dict,template="Regressor")
code_slow:
from tpot.config import regressor_config_dict
regressor_config_dict = {}
regressor_config_dict['xgboost.XGBRegressor'] = {
        'n_estimators': [100, 200, 300, 400, 500, 600, 700, 800, 900, 1000],
        'max_depth': range(3, 100),
        'learning_rate': [0.01, 0.03, 0.05, 0.07, 0.1, 0.3, 0.5],
        'subsample': np.arange(0, 1, 0.05),
        'gamma': np.arange(0, 10, 0.5),
        'min_child_weight': range(1, 10),
    }
tpot = TPOTRegressor(generations=20, population_size=50, verbosity=2, random_state=42,
                         n_jobs=-1, max_time_mins=5, config_dict=regressor_config_dict,template="Regressor")

**Comment by JDRomano2** [comment]
Hi @huangqingyi-code , the reason it's slow is because in the second block of code you are trying to build a machine learning pipeline using ONLY the `XGBRegressor` estimator. In the first (fast) block of code you have many other estimators available to TPOT, some of which train much more quickly than the gradient boosting regressor (like feature scalers and transformers, simpler estimators, etc.). This may also do feature selection, which means subsequent gradient boosting estimators will have fewer features and train more quickly.

**Comment by huangqingyi-code** [comment]
the hyper-parameter's combinations  of the model which in regressor_config_dict are small,for example     'xgboost.XGBRegressor': {
        'n_estimators': [100],
        'max_depth': range(1, 11),
        'learning_rate': [1e-3, 1e-2, 1e-1, 0.5, 1.],
        'subsample': np.arange(0.05, 1.01, 0.05),
        'min_child_weight': range(1, 21),
        'n_jobs': [1],
        'verbosity': [0],
        'objective': ['reg:squarederror']
    },
if I reset the hyper-parameter of the XGBRegressor as above  .Can I improve performance of the XGBRegressor ?
